File: 771-jewels-and-stones/main.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        
        s = {stone: stones.count(stone) for stone in set(stones)}
        j = [jewel for jewel in jewels]
        
        count = 0
        for jewel in j:
            if jewel in s:
                count += s[jewel]
        

        return count

File: 771-jewels-and-stones/main.rs
use std::collections::HashMap;

impl Solution {
    pub fn num_jewels_in_stones(jewels: String, stones: String) -> i32 {
        
        let mut stones_map: HashMap<char, i32> = HashMap::new();

        for stone in stones.chars() {
            *stones_map.entry(stone).or_insert(0) += 1;
        }

        let mut count: i32 = 0;
        for jewel in jewels.chars() {
            if let Some(&stone_count) = stones_map.get(&jewel) {
                count += stone_count;
            }
        }

        count
    }   
}
